research/flamby/find_best_hp.py

```diff
@@ -21,16 +21,7 @@ def get_weighted_loss_from_server_log(run_folder_path: str, experiment_name: str
     server_log_path = os.path.join(run_folder_path, "server.out")
     with open(server_log_path, "r") as handle:
         files_lines = handle.readlines()
-        if experiment_name == "fed_heart_disease":
-            line_to_convert = files_lines[-1].strip()
-        elif experiment_name == "fed_ixi":
-            line_to_convert = files_lines[-1].strip()
-        else:
-            if is_fenda:
-                # FENDA doesn't log the same way as other Fed-ISIC methods
-                line_to_convert = files_lines[-1].strip()
-            else:
-                line_to_convert = files_lines[-3].strip()
```

Comment:
I don't know why fenda was having this condition previously, but it raised error for me. So I needed to update it.

Reply:
If I remember correctly, I was getting a few extra lines appearing at the bottom of the logs that looked like this for models that we're FENDA

```
INFO flwr 2023-07-05 13:43:44,075 | server.py:146 | Best Aggregated (Weighted) Loss seen by the Server: 
4.862437355255427
Loaded pretrained weights for efficientnet-b0
Number of features output by EfficientNet 1280
```
It's possible that the logging changed a bit and these are being logged in their proper place now so that this message isn't popping up at the bottom anymore.

Reply:
It's also possible that it only appears at the bottom of the logs when you load the full EfficientNet model, which we do for all methods other than FENDA, FedPer, MOON, and PerFCL. So I think what we actually want to do to fix this is have these methods be considered as the "exceptions." Rather than the argument being `is_fenda` maybe we can change it to `is_fenda_fed_per_moon_perfcl`? It's long but covers all of the methods that don't load a full EfficientNet?

Reply:
Oh I see what you are talking about, I didn't see it as none of my methods used full EfficientNet models, so we should actually keep it and update this if to some other names. Thanks for pointing it out.